attestation/src/measurement.rs

```diff
@@ -149,9 +172,33 @@ impl<'a, D: Digest> AttestationManager<D> {
             );
         }
 
-        AttestationManager {
+        // Derive a TVM specific CDI from the TSM CDI.
+        let kdf = Hkdf::<D, H>::new(Some(&vm_id.to_le_bytes()), cdi);
+        let mut cdi_attest = [0u8; CDI_LEN];
+        kdf.expand(b"CDI_Attest", &mut cdi_attest)
+            .map_err(Error::InvalidCdiExpansion)?;
+
+        // Derive a reduced and salted CDI ID from the TVM specific CDI
+        let id_salt: [u8; 64] = [
+            // From the OpenDice implementation.
+            0xDB, 0xDB, 0xAE, 0xBC, 0x80, 0x20, 0xDA, 0x9F, 0xF0, 0xDD, 0x5A, 0x24, 0xC8, 0x3A,
+            0xA5, 0xA5, 0x42, 0x86, 0xDF, 0xC2, 0x63, 0x03, 0x1E, 0x32, 0x9B, 0x4D, 0xA1, 0x48,
+            0x43, 0x06, 0x59, 0xFE, 0x62, 0xCD, 0xB5, 0xB7, 0xE1, 0xE0, 0x0F, 0xC6, 0x80, 0x30,
+            0x67, 0x11, 0xEB, 0x44, 0x4A, 0xF7, 0x72, 0x09, 0x35, 0x94, 0x96, 0xFC, 0xFF, 0x1D,
+            0xB9, 0x52, 0x0B, 0xA5, 0x1C, 0x7B, 0x29, 0xEA,
```

Comment:
They are immutable and constant, but I'm expecting the function they live in to be rarely called (The attestation manager constructor, called once per VM lifetime. And the key_pair generation, called with each `GetEvidence` SBI request), so I figure it would make sense to have them on those rare stack frames.